src/relations/mysql.py

```diff
@@ -184,7 +184,7 @@ def _on_mysql_relation_created(self, event: RelationCreatedEvent) -> None:
             "host": primary_address,
             "password": password,
             "port": "3306",
-            "root_password": self.charm.app_peer_data["root-password"],
+            "root_password": self.charm.get_secret("app", "root-password"),
```

Comment:
legacy relation should continue using databag

Reply:
it is, the only change is that now we're getting the password with the `get_secret` helper instead of accessing the databag directly (which is how it should've been to begin with)